File: common/src/main/java/org/bread_experts_group/breadlib/network/payload/PayloadHandler.java
package org.bread_experts_group.breadlib.network.payload;

import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
import net.minecraft.world.entity.player.Player;
import org.bread_experts_group.breadlib.network.NetworkContext;

public interface PayloadHandler<T extends CustomPacketPayload> {
	void handle(T data, NetworkContext context);

	default void handle(T data, Player player) {
		this.handle(data, new NetworkContext(player));
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/network/payload/PayloadHandler.kt
package org.bread_experts_group.breadlib.network.payload

import net.minecraft.network.protocol.common.custom.CustomPacketPayload
import net.minecraft.world.entity.player.Player
import org.bread_experts_group.breadlib.network.NetworkContext

fun interface PayloadHandler<T : CustomPacketPayload> {
	fun handle(data: T, context: NetworkContext)

	fun handle(data: T, player: Player) {
		this.handle(data, NetworkContext(player))
	}
}
